experimental/framework/fluid-static/src/serviceAudience.ts

```diff
@@ -0,0 +1,139 @@
+  protected readonly audience: IAudience;
+
+  // Maintains a map of the last edited times keyed by client ID after the current client
+  // joined the session
+  // TODO: This will only maintain the list of edits made after the current client has joined.
+  // We would need an additional DataObject to support storing the last edited state from history
+  protected readonly lastEditedTimesByClient = new Map<string, Date>();
+
+  constructor(
+      protected readonly container: Container,
+      rootDataObject: RootDataObject,
+  ) {
+    super();
+    this.audience = container.audience;
+
+    // Consolidating both the addition/removal of members
+    this.audience.on("addMember", () => {
+      this.emit("membersChanged", this.getMembers());
+    });
+
+    this.audience.on("removeMember", () => {
+      this.emit("membersChanged", this.getMembers());
+    });
+
+    this.container.on("connected", () => {
+      this.emit("membersChanged", this.getMembers());
+    });
+
+    // Update the last edited information anytime the root object emits an op by a member
+    // of the current audience. This in turn emits a "lastEditedChanged" event so that
+    // any listeners can update their state to reflect the changes
+    rootDataObject.on("op", (message: ISequencedDocumentMessage) => {
+      this.lastEditedTimesByClient.set(message.clientId, new Date(message.timestamp));
+      const member = this.getMemberByClientId(
+        message.clientId,
+      );
+      if (member !== undefined) {
+        this.emit("lastEditedChanged", { member, timestamp :new Date(message.timestamp) });
+      }
+    });
+  }
+
+  /**
+   * @inheritdoc
+   */
+  public getMembers(): Map<string, IMember> {
+    const users = new Map<string, IMember>();
+    // Iterate through the members and get the user specifics.
+    this.audience.getMembers().forEach((member: IClient, clientId: string) => {
+      // Get all the current human members
+      if (member.details.capabilities.interactive) {
+        const userId = member.user.id;
+        if (users.has(userId)) {
+          const existingValue = users.get(userId);
+          if (existingValue) {
+            existingValue.connectedClients.push({
+              clientId,
+              connectionMode: member.mode,
+              timeLastActive: this.lastEditedTimesByClient.get(clientId),
+            });
+            existingValue.connectedClients.sort((a, b) =>
+              (b.timeLastActive?.getMilliseconds() ?? 0) - (a.timeLastActive?.getMilliseconds() ?? 0));
+          }
+        } else {
+          users.set(userId, {
+            userId,
+            connectedClients: [{
+              clientId,
+              connectionMode: member.mode,
+              timeLastActive: this.lastEditedTimesByClient.get(clientId),
+            }],
+          });
+        }
+      }
+    });
+    return users;
+  }
+
+  /**
+   * @inheritdoc
+   */
+  public getMyClient(): IConnectedClient | undefined {
```

Comment:
Should this exist here on the Audience?  I think the Container is the authority on clientId and connectionMode, Perhaps consulting with the (Fluid)Container directly for these items would be more straightforward.

Reply:
Yep, if we're removing the timeLastActive from here, I don't really see any further reason to have this be exposed as an `IConnectedClient` object here. We can expose it directly from the `FluidContainer`

Reply:
Removed! In a follow-up, perhaps we can restructure what is returned in FluidContainer to be an `IConnection` object holding the id and mode to allow for parity?